cli/cli.go

```diff
@@ -140,9 +142,42 @@ func checkTarget(cfg *Config) bool {
 	return true
 }
 
+func makeOptionMap(c *cli.Context, flags []cli.Flag, name, value string) map[string]interface{} {
+	omap := map[string]interface{}{name: value}
+	for _, flag := range flags {
+		switch f := flag.(type) {
+		case cli.StringFlag:
+			omap[f.Name] = c.String(f.Name)
+		case cli.BoolFlag:
+			omap[f.Name] = c.Bool(f.Name)
+		case cli.IntFlag:
+			omap[f.Name] = c.Int(f.Name)
+		}
```

Comment:
the error would be a panic since an unsupported type for an option string would be a programmer error. The only benefit I see is alerting the programmer sooner that it's an unsupported type -- rather than the option just not working. The downside is that it would be difficult to test the error path. What's your preference?